intg/src/main/java/org/apache/atlas/type/AtlasTypeRegistry.java

```diff
@@ -756,13 +757,21 @@ public void removeTypeByName(String name) throws AtlasBaseException {
 
         public List<AtlasBaseTypeDef> getDeleteedTypes() { return deletedTypes; }
 
+        void validateTypeCreation(AtlasBaseTypeDef typeDef) throws AtlasBaseException{
+            if(this.isRegisteredType(typeDef.getName()) && (this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.PRIMITIVE)||(this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.OBJECT_ID_TYPE)))){
```

Comment:
Why we are checking following ?
(this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.PRIMITIVE)||(this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.OBJECT_ID_TYPE)))

IMO, whenever creating any typeDef, do not allow in case Typedef.name OR its attributes.name are same a built in types' names